mycroft/util/network_utils.py

```diff
@@ -0,0 +1,77 @@
+import requests
+import socket
+from urllib.request import urlopen
+from urllib.error import URLError
+
+from .log import LOG
+
+
+def connected():
+    """Check connection by connecting to 8.8.8.8 and if google.com is
+    reachable if this fails, Check Microsoft NCSI is used as a backup.
+
+    Returns:
+        True if internet connection can be detected
+    """
+    if _connected_dns():
```

Comment:
It seems to me like the dns check is sufficient, and all the checks are potentially DoS and information disclosure concerns.

See [here](https://community.ntppool.org/t/recent-ntp-pool-traffic-increase/18/6) for an example of this type of problem in the wild.

Reply:
The checks have grown a bit over time. The dns lookup doesn't work on all networks due to filtering, hence ncis check...

The google check was added to handle some weird edge cases filtering out Mark-1's where the AP wasn't shut off completely. It should be removed at some point

Reply:
yeah, I guess I'm just not fully understanding why we'd use both domains here. Outbound port filtering makes sense for DNS vs HTTP protocols, I guess we're trying another tactic with http/https ports?

A "good internet citizen" would probably try to connect to a mycroft-operated or public for-this-purpose endpoint. NCSI seems to hit that (if intended for msft users), but dns and google.com don't strike me as good choices here.